### PR TITLE
ffda-wireless-rate-limiter: add package

### PR DESCRIPTION
**Pull request by blocktrron**
This adds a package which can be used to shape traffic on the wireless interfaces interface-wide or per-client.

### CODE REVIEW

File: ffda-wireless-rate-limiter/files/wireless-rate-limiter.upgrade.lua [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+local limits = wireless_limits_get()
+local limits_applied = false
+for _, value in pairs(limits) do
+	if wireless_limits_set(value.index, value.limits) then
+		limits_applied = true
+	end
+end
+
+-- Decide daemon necessity
+uci:set('wireless-rate-limiter', 'core', 'disabled', limits_applied and 0 or 1)
```
Comment (maurerle) [inline ffda-wireless-rate-limiter/files/wireless-rate-limiter.upgrade.lua:101]: what does this `and 0 or 1` do?
Looks like debug config to hard enforce/disable?

**Review by maurerle — COMMENTED** [review]
So this depends on installation of https://github.com/blocktrron/wireless-rate-limiter
Using `PKG_SOURCE_URL` as explained here: https://openwrt.org/docs/guide-developer/packages
Did not know about this feature before.

If this is configured through the `gluon` uci section, I would like to have this compatible with the other ingress limit?
https://github.com/freifunk-gluon/gluon/wiki/Commandline-administration#limit-bandwidth-gluon-v20191
Or is there a reason to name things `up` and `down` instead of the established `ingress` and `egress` option and `enabled`  instead of `disabled` as a config option?

The package TITLE seems to be copy-pasted from somewhere else and is not correct?

Other than that it looks good!

File: ffda-wireless-rate-limiter/files/wireless-rate-limiter.upgrade.lua (hunk 1 above)
Comment (blocktrron) [reply ffda-wireless-rate-limiter/files/wireless-rate-limiter.upgrade.lua:101]: This is just the elvis operator, you can negate the varname of `limits_applied` but i see this as more readable.

**Comment by blocktrron** [comment]
> I would like to have this compatible with the other ingress limit?

There is nothing to keep compatible, given they touch completely different areas and needs.

> Or is there a reason to name things up and down instead of the established ingress and egress option

Different package, so we don't need to be in style. Furthermore this does not limit the egress / ingress traffic on the interface but the rate towards the client, which is a per-station and not per-iface limit for the client-case.

> enabled instead of disabled

The disabled case when missing is equivalent to a False value. Chage to disabled and to retain the same logic the case would disable if the key is missing. Have a look at how this is handled for wifi-radios and wifi-interfaces on OpenWrt.

It makes sense from a yes / no standpoint if displayed in a UI, but this is just the flag for ignoring said section --> missing equals 0.

**Review by maurerle — APPROVED** [review]

**Pull request merged**